packages/storybook/src/executors/build-storybook/build-storybook.impl.ts

```diff
@@ -1,48 +1,109 @@
 } from '../utils';
 
 export interface StorybookBuilderOptions extends CommonNxStorybookConfig {
   quiet?: boolean;
   outputPath?: string;
-  docsMode?: boolean;
+  docsMode: boolean;
 }
-
 export default async function buildStorybookExecutor(
   options: StorybookBuilderOptions,
   context: ExecutorContext
 ) {
-  logger.info(`NX ui framework: ${options.uiFramework}`);
+  const storybook7 = isStorybookV7();
+
+  if (storybook7) {
+    storybookConfigExists(options.config, context.projectName);
+    const packageJson = readJsonFile(
+      path.join(workspaceRoot, 'package.json')
+    ) as PackageJson;
+    const buildOptions = {
+      ...options,
+      workspaceRoot: context.root,
+      configDir: options.config.configFolder,
+      packageJson,
+      watch: false,
+      mode: options?.['mode'] ?? 'static',
+      outputDir:
+        (options?.['outputDir'] || options?.['output-dir']) ??
+        options.outputPath,
+      ignorePreview: options['ignorePreview'] ?? false,
+      cache: options['cache'] ?? false,
+    } as CLIOptions &
+      LoadOptions &
+      BuilderOptions & {
+        outputDir: string;
+      };
 
-  const frameworkPath = getStorybookFrameworkPath(options.uiFramework);
-  const { default: frameworkOptions } = await import(frameworkPath);
+    logger.info(`NX Storybook builder starting ...`);
+    await runInstance(buildOptions);
+    logger.info(`NX Storybook builder finished ...`);
+    logger.info(`NX Storybook files available in ${options.outputPath}`);
+    return { success: true };
+  } else {
+    logger.info(`NX ui framework: ${options.uiFramework}`);
 
-  const option = storybookOptionMapper(options, frameworkOptions, context);
+    const frameworkPath = getStorybookFrameworkPath(options.uiFramework);
+    const { default: frameworkOptions } = await import(frameworkPath);
 
-  // print warnings
-  runStorybookSetupCheck(options);
+    const option = storybookOptionMapper(
+      options,
+      frameworkOptions,
+      context
+    ) as CLIOptions &
+      LoadOptions &
+      BuilderOptions & {
+        outputDir: string;
+      };
 
-  logger.info(`NX Storybook builder starting ...`);
-  await runInstance(option);
-  logger.info(`NX Storybook builder finished ...`);
-  logger.info(`NX Storybook files available in ${options.outputPath}`);
-  return { success: true };
+    // print warnings
+    runStorybookSetupCheck(options);
+
+    logger.info(`NX Storybook builder starting ...`);
+    await runInstance(option);
+    logger.info(`NX Storybook builder finished ...`);
+    logger.info(`NX Storybook files available in ${options.outputPath}`);
+    return { success: true };
+  }
 }
 
-function runInstance(options: StorybookBuilderOptions): Promise<void> {
-  const env = process.env.NODE_ENV ?? 'production';
+function runInstance(
+  options: CLIOptions &
+    LoadOptions &
+    BuilderOptions & {
+      outputDir: string;
+    }
+): Promise<void> {
+  const env = process.env.NODE_ENV ?? 'PRODUCTION';
```

Comment:
This should be `'production'` right? I've always seen it used with lowercase in projects.

Like this?

```
process.env.NODE_ENV ??= 'production'
```


Do we need `configType` in `buildStaticStandalone` to be the same as `process.env.NODE_ENV`? It can be out of sync now (although in the build case it should always be `production`).